kofta/src/app/components/KeybindListener.tsx

```diff
@@ -1,22 +1,91 @@
-import React, { useMemo } from "react";
+import React, { useMemo, useEffect } from "react";
 import { GlobalHotKeys } from "react-hotkeys";
 import { wsend } from "../../createWebsocket";
-import { useKeyMapStore } from "../../webrtc/stores/useKeyMapStore";
+import { useKeyMapStore, REQUEST_TO_SPEAK_KEY, MUTE_KEY, CHAT_KEY, INVITE_KEY } from "../../webrtc/stores/useKeyMapStore";
 import { useMuteStore } from "../../webrtc/stores/useMuteStore";
 import { useRoomChatStore } from "../modules/room-chat/useRoomChatStore";
 import { useHistory } from "react-router-dom";
 import { modalConfirm } from "../components/ConfirmModal";
+import isElectron from "is-electron";
 
-interface KeybindListenerProps {}
+let ipcRenderer: any = undefined;
+if (isElectron()) {
+  ipcRenderer = window.require('electron').ipcRenderer
+}
 
-export const KeybindListener: React.FC<KeybindListenerProps> = ({}) => {
+interface KeybindListenerProps { }
+
+export const KeybindListener: React.FC<KeybindListenerProps> = ({ }) => {
+
+  return (
+    <>
+      {isElectron() ? <ListenerElectron /> : <ListenerBrowser />}
+    </>
+  )
+
+};
+
+function ListenerElectron() {
   const { keyMap } = useKeyMapStore();
-	const [toggleOpen, newUnreadMessages] = useRoomChatStore((s) => [
-		s.toggleOpen,
-		s.newUnreadMessages,
-	]);
+  const [toggleOpen, newUnreadMessages] = useRoomChatStore((s) => [
+    s.toggleOpen,
+    s.newUnreadMessages,
+  ]);
   const history = useHistory();
+  useEffect(() => {
+    ipcRenderer.on(REQUEST_TO_SPEAK_KEY, async (event: any, args: any) => {
+      modalConfirm("Would you like to ask to speak?", () => {
+        wsend({ op: "ask_to_speak", d: {} });
+      });
+    })
+    ipcRenderer.on(MUTE_KEY, async (event: any, args: any) => {
+      const { muted, setMute } = useMuteStore.getState();
+      wsend({
+        op: "mute",
+        d: { value: !muted },
+      });
+      setMute(!muted);
+    })
+    ipcRenderer.on(INVITE_KEY, async (event: any, args: any) => {
+      wsend({ op: "fetch_invite_list", d: { cursor: 0 } });
+      history.push("/invite");
+    })
+    ipcRenderer.on(CHAT_KEY, async (event: any, args: any) => {
+      toggleOpen();
+    })
```

Comment:
do you need to unsubscribe?

Reply:
Umm i dont think there's any need for it but electron does provide a function to remove the listener, so ig ill put that in cleanup. Will push a new commit soon with the PTT key even working